src/clj/puppetlabs/services/legacy_routes/legacy_routes_service.clj

```diff
@@ -21,7 +21,14 @@
           puppet-version (get-in config [:puppet-server :puppet-version])
           ca-settings (ca/config->ca-settings (get-config))
           ca-mount (get-route (tk-services/get-service this :CaService))
-          master-mount (master-service/get-master-path config)
+          master-ns (keyword (tk-services/service-symbol
+                               (tk-services/get-service this :MasterService)))
+          master-route-config (master-core/get-master-route-config
```

Comment:
I don't think so.  IIRC, the current routing service would allow you to do a `get-route` for an arbitrary service - like is done for the `:CaService` above.  We still have the issue here, though, where the route payload for the `:MasterService` might be a map with the route value as one of the route ids vs. just being a string, and the `get-route` implementation forces you to specify in advance which one you want, throwing an `Exception` if the expectation isn't met.  I think this is why you filed https://tickets.puppetlabs.com/browse/TK-223, right?  To avoid having to write code that does up to two `get-route` calls and handles `Exception`s on failure, this PR basically preserves the logic that was in the implementation previously.

The only changes here are that the service symbol is determined dynamically from whatever service is found which implements the `:MasterService` protocol, and the implementation was moved down into the master core namespace.  The only reason I moved it into the core namespace is that it felt a little cleaner for it to be there given that we don't otherwise want to have any direct coupling between the `legacy_routes` service and the OSS `master` service since the OSS `master` service isn't even loaded in PE.